src/api/src/resource/mod.rs

```diff
@@ -35,6 +35,18 @@ pub struct ResourceDesc {
     pub resource_tag: String,
 }
 
+impl ResourceDesc {
+    pub fn check_legality(&self) -> Result<()> {
+        if &self.repository_name == "." || &self.repository_name == ".." {
+            bail!(anyhow!("Illegal resource repository name"));
```

Comment:
Similar to the previous comment:

```suggestion
            bail!(anyhow!("Invalid resource repository name"));
```